app/inpututils.cpp

```diff
@@ -928,6 +928,38 @@ QString InputUtils::geometryFromLayer( QgsVectorLayer *layer )
   return QString();
 }
 
+qreal InputUtils::calculateScreenDpr()
+{
+  const QWindowList windows = QGuiApplication::topLevelWindows();
+
+  if ( !windows.isEmpty() )
+  {
+    QScreen *screen = windows.at( 0 )->screen();
+    double dpiX = screen->physicalDotsPerInchX();
+    double dpiY = screen->physicalDotsPerInchY();
```

Comment:
Hmm actually this is already giving the scaled DPI (e.g. 140.67 on my phone instead of the actual 422)

I guess we need to multiply it by `screen->devicePixelRatio()`

Reply:
That is correct